exercises/webapp/src/components/HackerNews.vue

```diff
@@ -1,33 +1,42 @@
 <template>
   <div class="hacker-news">
     <h1>News List</h1>
-    <div v-for="item in sortedItems" :key="item.id">
-      <NewsItem v-bind:news-item="item" v-on:news_remove="onNewsRemove" v-on:update="onVoteChange"/>
+    <div v-if="sortedItems.length">
+      <div v-for="item in sortedItems" :key="item.id">
+        <NewsItem v-bind:news-item="item" v-on:news-remove="onNewsRemove" v-on:update="onVoteChange"/>
+      </div>
     </div>
-    <NewsForm v-on:news_add="onNewsAdd"></NewsForm>
+    <div v-else id="news-placeholder">
+      The list is empty :(
+    </div>
+    <NewsForm v-on:news-add="onNewsAdd" v-on:switch="onSwitch"></NewsForm>
   </div>
 </template>
 
 <script>
-import NewsItem from './NewsItem.vue';
-import NewsForm from './NewsForm.vue';
+import NewsItem from "./NewsItem.vue";
+import NewsForm from "./NewsForm.vue";
 
 export default {
-  name: 'HackerNews',
+  name: "HackerNews",
   components: {NewsItem, NewsForm},
+  props: ["initialNews"],
   data() {
     return {
-      items: [{
-        id: 0,
-        title: "Start Message",
-        voteCount: 0,
-      }],
+      ascending: false,
+      items: [...this.initialNews],
       id: 1
```

Comment:
You might run into a situation where you have duplicate keys. You could make this a computed property:
```js
computed: {
  nextId() {
    return Math.max(...this.items.map(item => item.id), 0) + 1
  }
}
```